src/main/java/at/hannibal2/skyhanni/features/misc/halloweenlobbywaypoints/BasketWaypoints.kt

```diff
@@ -0,0 +1,61 @@
+package at.hannibal2.skyhanni.features.misc.halloweenlobbywaypoints
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.ScoreboardData
+import at.hannibal2.skyhanni.test.GriffinUtils.drawWaypointFilled
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.RenderUtils.drawDynamicText
+import net.minecraft.client.Minecraft
+import net.minecraftforge.client.event.RenderWorldLastEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class BasketWaypoints {
+    private val config get() = SkyHanniMod.feature.misc.halloweenBasket
+    private var waypoint: LorenzVec? = null
+    private var waypointName: String? = null
+
+    @SubscribeEvent
+    fun onRenderWorld(event: RenderWorldLastEvent) {
+        if (LorenzUtils.inSkyBlock || !LorenzUtils.onHypixel) return
+
+        if (!chechScoreboardHalloweenSpecific()) return
+
+        if (config.allWaypoints) {
+            for (basket in Basket.entries) {
+                event.drawWaypointFilled(basket.waypoint, LorenzColor.GOLD.toColor())
+                event.drawDynamicText(basket.waypoint, "§6" + basket.basketName, 1.5)
+            }
+        }
+
+        if (config.allEntranceWaypoints) {
+            for (basketEntrance in BasketEntrances.entries) {
+                event.drawWaypointFilled(basketEntrance.waypoint, LorenzColor.YELLOW.toColor())
+                event.drawDynamicText(basketEntrance.waypoint, "§e" + basketEntrance.basketEntranceName, 1.5)
+            }
+            return
+        }
+
+        if (LorenzUtils.skyBlockArea == "?") return
+
+        waypoint?.let {
+            event.drawWaypointFilled(it, LorenzColor.GOLD.toColor())
+            event.drawDynamicText(it, "§6" + waypointName!!, 1.5)
+        }
+    }
+
+    private fun chechScoreboardHalloweenSpecific(): Boolean {
+        return (
+            ( ScoreboardData.sidebarLinesFormatted.any {
+                it.contains("Hypixel Level")
+            } ) && 
+            ( ScoreboardData.sidebarLinesFormatted.any {
+                it.contains("Halloween ")
+            } ) && 
+            ( ScoreboardData.sidebarLinesFormatted.any {
+                it.contains("Baskets ")
```

Comment:
You can use `list.any { siA && isB && isC}` instead of 3 `any` blocks

Reply:
will do in a moment eating brunch rn

Reply:
@hannibal002 
tried it, failed. it seems that this only works if we're checking bits and pieces of one specific line in the scoreboard

Reply:
Yeah, my bad, this was wrong from me. My approach does obviously not work if those strings aren't all in the same line.